core/src/main/c/share/net.c

```diff
@@ -239,7 +239,17 @@ JNIEXPORT jint JNICALL Java_io_questdb_network_Net_configureLinger
 
 JNIEXPORT jint JNICALL Java_io_questdb_network_Net_connect
         (JNIEnv *e, jclass cl, jint fd, jlong sockAddr) {
-    return connect((int) fd, (const struct sockaddr *) sockAddr, sizeof(struct sockaddr));
+    jboolean retry = 1;
```

Comment:
`retry` is initially set to 1, and the bitwise OR assignment (|=) keeps it as 1 throughout the loop. Wouldn't using `RESTARTABLE` be enough to handle retry?

Reply:
That's a bug, retry should be initiated as 0, it's fixed now.

If there were no retries `errno == EISCONN` is an error, if there was a retry then this is OK. I don't see how `RESTARTABLE` can handle that